### PR TITLE
Fix a fuzz issue with #6984

### PR DESCRIPTION
**Pull request by kripken**
When I refactored the optimizeDroppedBlock logic in #6982, I didn't move the
unreachability check with that code, which was wrong. When that function
was called from another place in #6984, the fuzzer found an issue.

Diff without whitespace is smaller.

The testcase is a little bizarre, but it's the one the fuzzer found and I can't
find a way to generate a better one (other than to reduce it, which I did).

### CODE REVIEW

**Comment by kripken** [comment]
Actually the test update here reverts almost all the test updates from #6984 - those changes were on blocks with unreachable children. The change was safe on them, but in general removing a block value in the presence of unreachable code is tricky, so it's best to avoid it.

File: test/lit/passes/gufa-optimizing_merge-blocks_simplify-locals-notee.wast [hunk 1]
```diff
@@ -0,0 +1,58 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; RUN: wasm-opt %s --gufa-optimizing  --merge-blocks --simplify-locals-notee -all -S -o - | filecheck %s
+;;
+;; GUFA manages to generate IR that makes merge-blocks leave a stale type, if we
+;; are not careful. simplify-locals-notee then errors if that stale type is
+;; present, so all the passes are needed to hit this bizarre corner case.
```
Comment (aheejin) [inline test/lit/passes/gufa-optimizing_merge-blocks_simplify-locals-notee.wast:6]: Not sure what this means... Why do all the passes need to hit this corner case wheh simplify-lcoals-notee errors out?

**Review by aheejin — APPROVED** [review]

File: test/lit/passes/gufa-optimizing_merge-blocks_simplify-locals-notee.wast (hunk 1 above)
Comment (kripken) [reply test/lit/passes/gufa-optimizing_merge-blocks_simplify-locals-notee.wast]: It isn't that the passes each need to do that, but that the testcase must include all those passes, for it to hit the error. I improved the text now.

**Comment by kripken** [comment]
Hmm, the testcase now errors on CI for some reason I can't figure out. As it was a very low-value testcase anyhow (with that weird sequence of passes, it is hard to actually understand what's going wrong), I removed it.

**Pull request merged**